tests/integration/targets/aws_secret/tasks/replication.yml

```diff
@@ -0,0 +1,78 @@
+---
+- block:
+  # ============================================================
+  # Creation/Deletion testing
+  # ============================================================
+  - name: add secret to AWS Secrets Manager
+    aws_secret:
+      name: "{{ secret_name }}"
+      state: present
+      secret_type: 'string'
+      secret: "{{ super_secret_string }}"
+      replica:
+       - region: 'us-east-2'
+       - region: 'us-west-2'
+         kms_key_id: 'alias/aws/secretsmanager'
+    register: result
+
+  - name: assert correct keys are returned
+    assert:
+      that:
+        - result.changed
+        - result.arn is not none
+        - result.name is not none
+        - result.replication_status[0]["region"] == 'us-east-2'
+        - result.replication_status[1]["region"] == 'us-west-2'
+        - result.replication_status[1]["kms_key_id"] == 'alias/aws/secretsmanager'
+        - result.tags is not none
+        - result.version_ids_to_stages is not none
+
+  - name: no changes to secret
```

Comment:
I came to a new funny idea :) 

```yml
  - name: secret 2 replicas
    aws_secret:
      name: "{{ secret_name }}"
      state: present
      secret_type: 'string'
      secret: "{{ super_secret_string }}"
      replica:
       - region: 'us-east-2'
       - region: 'us-west-2'
         kms_key_id: 'alias/aws/secretsmanager'
    register: result

  - name: change replicas
    aws_secret:
      name: "{{ secret_name }}"
      state: present
      secret_type: 'string'
      secret: "{{ super_secret_string }}"
      replica:
       - region: 'us-east-2'
       - region: 'eu-central-1'
         kms_key_id: 'alias/aws/secretsmanager'
    register: result
```

what will happen?  
Will `us-west-2` still be remain or be removed?  
Do we need a boolean `purge_replica` parameter that controlls the list. (_remove diff? or just append?_)

anymore opinions about this? cc @alinabuzachis @jillr @tremble

Reply:
I think a "purge_replicas" option which defaults to True is probably the best option.  It makes things more declarative.

Reply:
I agree to have `purge_replica `.

Reply:
Currently what will happen is compare_regions() will add `us-west-2` to the list of regions to be removed and `eu-central-1` to the list of regions to be added.  So this will be true:

```
  - name: assert that replica regions changed
    assert:
      that:
        - not result.failed
        - result.replication_status[0]["region"] == 'us-east-2'
        - result.replication_status[1]["region"] == 'eu-central-1'
        - result.replication_status[1]["kms_key_id"] == 'alias/aws/secretsmanager'
```

I also added that integration test.

Reply:
@emillbrandt-ngt sorry for the late delay.  
the `purge_*` terminology is widespread in ansible modules.  
It would be great if you find some time to implement it. It think it is not a big deal, because you only have to remove items in `state: present` when `purge_replicas` is also requested.

Reply:
I've updated this so that replicas are only purged if someone explicitly passes the replica argument.  I think this is reasonable for now.

As the PR was originally, running this against a secret which someone had manually replicated (before the module supported replication) would have nuked the replication.